front/app/components/CampaignConsentForm/index.tsx

```diff
@@ -76,6 +81,16 @@ const CampaignConsentForm = ({
     if (!isNilOrError(originalCampaignConsents)) {
       const campaignConsentsEntries = originalCampaignConsents.data
         .sort((a, b): number => a.id.localeCompare(b.id))
+        .filter((consent) => {
+          if (
+            (internalCommentTypes as unknown as CampaignName).includes(
```

Comment:
Not the greatest fan of this TS construct but not sure how to solve it right away and whether it's worth the time (code will likely not be touched very often).

Reply:
That said, we should pass a parameter to the BE so this filtering is done there. Similar to how we pass withoutCampaignNames to useCampaigns.

Reply:
> Not the greatest fan of this TS construct 

Not sure how to address this either. In my opinion, TS shouldn't be complaining about this. `internalCommentTypes` are part of `CampaignName` which includes other names, and `internalCommentTypes` is a "subset" of `CampaignName`. Irrespective of what values we get on either side, the includes will give us a boolean. The unknown is supposed to be for the bits that don't intersect but that doesn't matter in this case.

> That said, we should pass a parameter to the BE so this filtering is done there. Similar to how we pass withoutCampaignNames to useCampaigns.

On the BE side, I thought about it but then I thought that would delay us on internal commenting further. Tagging @jinjagit to assess the impact of BE work when he gets back. If it is direct, we can add the BE work here and I'll make the FE changes

Reply:
It should be simple enough to filter out the `internal_comments` campaigns, given a request parameter.

What parameter would you like to pass and with what value (to indicate filtering out the `internal_comments` campaigns)?

Reply:
Ahh, I see. We already have that.

Reply:
Yes, we have `without_campaign_names` for the campaigns endpoint. We can do the same

Reply:
I mean, I think the BE functionality is already in place. You just need to pass the right values to filter out campaigns. as desired.

I'm going to check it works, and test what those values should be.

Reply:
Weird that it isn't in the documentation [here](https://developers.citizenlab.co/api-docs/ee/frontweb_api/master/campaign_consents/list_all_campaign_consents.html)

Reply:
You want to filter the `campaigns` endpoint or `campaigns_consents`?

Reply:
Both, the campaigns works well with `without_campaign_names`. We need the same for `campaigns_consents`

Reply:
Basic filtering of `campaigns_consents` added in this PR (WIP - I still need to add tests). You can pull it in to your branches already, if that would help.